config/initializers/secure_headers.rb

```diff
@@ -62,7 +62,7 @@
 SecureHeaders::Configuration.override(:saml) do |config|
   provider_attributes = SERVICE_PROVIDERS.values
 
-  acs_urls = provider_attributes.map { |hash| hash['acs_url'] }
+  acs_urls = provider_attributes.map { |hash| hash['acs_url'] }.compact
```

Comment:
To give some background on this, in case you're not familiar with what this does, the `acs_url` is the URL on the SP side where the IdP is supposed to redirect. By default, our CSP (Content Security Policy) settings only allow form submissions to `self`, i.e. the current IdP domain name. To allow other URLs, we parse all the `acs_urls` in `service_providers.yml` at runtime, then add them to the `form_action` array. In addition, we also need to tell `saml_idp_controller` to use this override.

I'm assuming the form_action restrictions will also apply to OpenId Connect, but maybe I'm wrong? Have you tested this?